shuup/admin/modules/users/views/detail.py

```diff
@@ -208,13 +217,22 @@ def _build_existing_user(self):
             getattr(current_user, "is_superuser", False) or
             getattr(current_user, "is_staff", False)
         )
-        if (is_current_user_superuser_or_staff and get_front_url() and
-                user.is_active and not user.is_superuser and not user.is_staff):
+        can_impersonate = bool(
+            is_current_user_superuser_or_staff and user.is_active and not user.is_superuser
+        )
+
+        if (can_impersonate and get_front_url() and not user.is_staff):
             self.append(PostActionButton(
                 post_url=reverse("shuup_admin:user.login-as", kwargs={"pk": user.pk}),
                 text=_(u"Login as User"),
                 extra_css_class="btn-gray"
             ))
+        elif can_impersonate and get_admin_url():
```

Comment:
I am not sure I got this. Just because there is admin url means the user can impersonate as staff?
Also, do you think we need two buttons? Can we have just one? If the user is staff then redirect to admin, otherwise redirect to front (using those urls in settings)

Reply:
I think two buttons is better for permissions. Maybe someone should only impersonate end users and impersonating staff users should only be for some main guys.

Reply:
Admin url logic is same as with front so if for some reason we do not know the redirect url then let's not show te button.

Reply:
got it, makes sense